src/background/actions/issue-filing-action-creator.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { autobind } from '@uifabric/utilities';
+
+import { RegisterTypeToPayloadCallback } from '../../common/message';
+import { Messages } from '../../common/messages';
+import * as TelemetryEvents from '../../common/telemetry-events';
+import { BrowserAdapter } from '../browser-adapter';
+import { TelemetryEventHandler } from '../telemetry/telemetry-event-handler';
+import { DetailsViewController } from './../details-view-controller';
+import { BaseActionPayload, OpenIssueFilingSettingsDialogPayload, OpenNewWindowPayload } from './action-payloads';
+import { IssueFilingActions } from './issue-filing-actions';
+
+export class IssueFilingActionCreator {
+    constructor(
+        private issueFilingActions: IssueFilingActions,
+        private browserAdapter: BrowserAdapter,
+        private telemetryEventHandler: TelemetryEventHandler,
+        private registerTypeToPayloadCallback: RegisterTypeToPayloadCallback,
+        private detailsViewController: DetailsViewController,
+    ) {}
+
+    public registerCallbacks(): void {
+        this.registerTypeToPayloadCallback(Messages.IssueFiling.OpenDialog, this.onOpenIssueFilingDialog);
```

Comment:
call it openSettingsDialog & closeSettingsDialog.
also rename onOpenIssueFilingDialog to onOpenIssueFilingSettingsDialog